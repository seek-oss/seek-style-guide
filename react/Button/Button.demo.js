import React from 'react';
import { Button, HeartIcon, StarIcon } from 'seek-style-guide/react';
import * as sketch from './Button.sketch';
import styles from './Button.less';
import classnames from 'classnames';

export default {
  route: '/button',
  title: 'Button',
  category: 'Form',
  component: Button,
  sketch,
  initialProps: {
    children: 'Hello world',
    color: 'pink'
  },
  options: [
    {
      label: 'States',
      type: 'checklist',
      states: [
        {
          label: 'Hover',
          transformProps: ({ className, ...props }) => ({
            ...props,
            className: classnames(className, styles.rootHover)
          })
        },
        {
          label: 'Active',
          transformProps: ({ className, ...props }) => ({
            ...props,
            className: classnames(className, styles.rootActive)
          })
        },
        {
          label: 'Focus',
          transformProps: ({ className, ...props }) => ({
            ...props,
            className: classnames(className, styles.rootFocus)
          })
        },
        {
          label: 'Loading',
          transformProps: props => ({
            ...props,
            loading: true
          })
        },
        {
          label: 'Full width',
          transformProps: props => ({
            ...props,
            fullWidth: true
          })
        },
        {
          label: 'Ghost',
          transformProps: props => ({
            ...props,
            ghost: true
          })
        }
      ]
    },
    {
      label: 'Component',
      type: 'radio',
      states: [
        {
          label: 'As Button',
          transformProps: props => ({
            ...props,
            component: 'button'
          })
        },
        {
          label: 'As Link',
          transformProps: props => ({
            ...props,
            component: 'a',
            href: '#'
          })
        },
        {
          label: 'As Div',
          transformProps: props => ({
            ...props,
            component: 'div'
          })
        }
      ]
    },
    {
      label: 'Colour',
      type: 'radio',
      states: [
        {
          label: 'Pink',
          transformProps: props => ({
            ...props,
            color: 'pink'
          })
        },
        {
          label: 'White',
          transformProps: props => ({
            ...props,
            color: 'white'
          })
        },
        {
          label: 'Blue',
          transformProps: props => ({
            ...props,
            color: 'blue'
          })
        },
        {
          label: 'Gray',
          transformProps: props => ({
            ...props,
            color: 'gray'
          })
        },
        {
          label: 'Transparent',
          transformProps: props => ({
            ...props,
            color: 'transparent'
          })
        }
      ]
    },
    {
      label: 'Icon',
      type: 'radio',
      states: [
        {
          label: 'No Icon',
          transformProps: props => props
        },
        {
          label: 'Heart Icon',
          transformProps: props => ({
            ...props,
            children: (
              <span>
                <HeartIcon filled={true} />
                &nbsp;Click here
              </span>
            )
          })
        },
        {
          label: 'Star Icon',
          transformProps: props => ({
            ...props,
            children: (
              <span>
                <StarIcon filled={true} />
                &nbsp;Click here
              </span>
            )
          })
        }
      ]
    }
  ]
};
