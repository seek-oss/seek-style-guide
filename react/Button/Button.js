// @flow
import styles from './Button.less';
import React from 'react';
import { Component } from 'react';
import classnames from 'classnames';
import capitalize from 'lodash/capitalize';

type Props = {
  color: 'pink' | 'blue' | 'gray' | 'transparent' | 'white',
  children: React$Node,
  className?: string,
  component?: string | Function,
  ghost?: boolean,
  loading?: boolean,
  fullWidth?: boolean
};

export default class Button extends Component<Props> {
  static displayName = 'Button';

  static defaultProps = {
    className: '',
    ghost: false,
    loading: false,
    fullWidth: false,
    component: 'button'
  };

  button: ?HTMLElement;
  props: Props;

  render() {
    const {
      color,
      ghost,
      className,
      loading,
      fullWidth,
      children,
      component = 'button',
      ...restProps
    } = this.props;

    const combinedProps = {
      className: classnames(styles.root, className, {
        [styles.loading]: loading,
        [styles.fullWidth]: fullWidth,
        [styles.root_isGhost]: ghost && color !== 'transparent',
        [styles[`root_is${capitalize(color)}`]]: color
      }),
      disabled: loading,
      ...restProps
    };

    return React.createElement(component, combinedProps, children);
  }
}
