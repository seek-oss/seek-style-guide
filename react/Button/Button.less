@import (reference) "~seek-style-guide/theme";

.hoverState(@rules) {
  &:hover,
  &.rootHover {
    @rules();
  }
}

.activeState(@rules) {
  &:active,
  &.rootActive {
    @rules();
  }
}

.focusState(@rules) {
  &:focus,
  &.rootFocus {
    @rules();
  }
}

.buttonColor(@color, @background-color) {
  color: @color;
  background-color: @background-color;

  .hoverState({
    background-color: lighten(@background-color, 5%);
  });

  .activeState({
    background-color: darken(@background-color, 5%);
  });
}

.root {
  .touchableText();
  display: inline-block;
  text-decoration: none;
  user-select: none;
  text-align: center;
  padding: 0 @grid-gutter-width;
  box-shadow: 0 1px fade(@sk-black, 70%);
  border: 0;
  border-radius: 2px;
  transition: background-color 0.2s, transform .2s cubic-bezier(0.020, 1.505, 0.745, 1.235);
  cursor: pointer;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  .activeState({
    box-shadow: none;
    outline: none;
    transform: scale(0.95);
  });
  .focusState({
    .focus();
  });
}

.root_isPink {
  .buttonColor(@sk-white, @sk-pink);
}

.root_isBlue {
  .buttonColor(@sk-white, @sk-highlight);
}

.root_isGray {
  .buttonColor(@sk-black, @sk-mid-gray-light);
}

.root_isTransparent {
  color: @sk-link;
  background-color: transparent;
  box-shadow: none;
  padding-left: 0;
  padding-right: 0;

  .hoverState({
    text-decoration: underline;
  });

  .activeState({
    transform: none;
  });

  @media only screen and (min-width: @responsive-breakpoint) {
    .touchableText(@standard-type-scale);
  }
}

.loading {
  padding-right: @grid-gutter-width * 2;

  &:after {
    content: ' \2026';
    text-align: left;
    width: 0;
    overflow-x: hidden;
    display: inline-block;
    position: absolute;
    animation-duration: 0.8s;
    animation-name: loading;
    animation-iteration-count: infinite;
  }
}

.fullWidth {
  width: 100%;
}

@keyframes loading {
  to {
    width: @grid-gutter-width;
  }
}
