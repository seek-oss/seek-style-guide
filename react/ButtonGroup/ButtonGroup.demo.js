import React from 'react';
import PropTypes from 'prop-types';
import {
  ButtonGroup,
  Button,
  PageBlock,
  Section
} from 'seek-style-guide/react';
import * as sketch from './ButtonGroup.sketch';

const ButtonGroupContainer = ({ component: DemoComponent, componentProps }) => (
  <PageBlock>
    <Section>
      <DemoComponent {...componentProps} />
    </Section>
  </PageBlock>
);
ButtonGroupContainer.propTypes = {
  component: PropTypes.any,
  componentProps: PropTypes.object.isRequired
};

export default {
  route: '/button-group',
  title: 'Button Group',
  category: 'Form',
  component: ButtonGroup,
  container: ButtonGroupContainer,
  block: true,
  sketch,
  initialProps: {
    children: <Button color="pink">Create a Profile</Button>
  },
  options: [
    {
      label: 'Number of Buttons',
      type: 'radio',
      states: [
        {
          label: 'One Button',
          transformProps: props => ({
            ...props,
            children: <Button color="pink">Create a Profile</Button>
          })
        },
        {
          label: 'Two Buttons',
          transformProps: props => ({
            ...props,
            children: [
              <Button key="1" color="pink">
                Create a Profile
              </Button>,
              <Button key="2" color="transparent">
                Cancel
              </Button>
            ]
          })
        },
        {
          label: 'Three Buttons',
          transformProps: props => ({
            ...props,
            children: [
              <Button key="1" color="pink">
                Create a Profile
              </Button>,
              <Button key="2" color="transparent">
                Clear
              </Button>,
              <Button key="3" color="transparent">
                Cancel
              </Button>
            ]
          })
        }
      ]
    }
  ]
};
