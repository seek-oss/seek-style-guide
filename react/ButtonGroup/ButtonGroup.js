import styles from '../Button/Button.less';
import React from 'react';
import PropTypes from 'prop-types';

import classnames from 'classnames';

export default function ButtonGroup({ className, children, ...restProps }) {
  return (
    <div
      {...restProps}
      className={classnames({ [className]: className, [styles.group]: true })}
    >
      {children}
    </div>
  );
}

ButtonGroup.propTypes = {
  className: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
};
