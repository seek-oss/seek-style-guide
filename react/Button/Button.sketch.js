import React from 'react';
import Button from './Button';

export const symbols = {
  'Button/Pink': <Button color="pink">Button</Button>,
  'Button/Pink Ghost': (
    <Button color="pink" ghost>
      Button
    </Button>
  ),
  'Button/Blue': <Button color="blue">Button</Button>,
  'Button/Blue Ghost': (
    <Button color="blue" ghost>
      Button
    </Button>
  ),
  'Button/Gray': <Button color="gray">Button</Button>,
  'Button/Gray Ghost': (
    <Button color="gray" ghost>
      Button
    </Button>
  ),
  'Button/Transparent': <Button color="transparent">Button</Button>,
  'Button/White Ghost': (
    <Button color="white" ghost>
      Button
    </Button>
  )
};
