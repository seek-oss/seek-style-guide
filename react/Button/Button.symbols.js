import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Button from './Button';

export default class ButtonSymbols extends Component {
  propTypes = {
    SketchSymbol: PropTypes.func.isRequired
  };

  render() {
    const { SketchSymbol } = this.props;

    return (
      <div>
        <SketchSymbol name="Button/pink">
          <Button color="pink">Button</Button>
        </SketchSymbol>

        <SketchSymbol name="Button/blue">
          <Button color="blue">Button</Button>
        </SketchSymbol>

        <SketchSymbol name="Button/transparent">
          <Button color="transparent">Button</Button>
        </SketchSymbol>
      </div>
    );
  }
}
