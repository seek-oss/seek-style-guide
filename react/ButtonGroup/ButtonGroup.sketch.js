import React from 'react';
import Button from '../Button/Button';
import ButtonGroup from './ButtonGroup';

export const symbols = {
  'Button Group/Pink': (
    <ButtonGroup>
      <Button color="pink">Button 1</Button>
    </ButtonGroup>
  ),
  'Button Group/Pink Ghost': (
    <ButtonGroup>
      <Button color="pink" ghost>
        Button 1
      </Button>
    </ButtonGroup>
  ),
  'Button Group/Blue Ghost': (
    <ButtonGroup>
      <Button color="blue" ghost>
        Button 1
      </Button>
    </ButtonGroup>
  ),
  'Button Group/Pink + Blue': (
    <ButtonGroup>
      <Button color="pink">Button 1</Button>
      <Button color="blue">Button 2</Button>
    </ButtonGroup>
  ),
  'Button Group/Pink + Blue + Transparent': (
    <ButtonGroup>
      <Button color="pink">Button 1</Button>
      <Button color="blue">Button 2</Button>
      <Button color="transparent">Button 3</Button>
    </ButtonGroup>
  ),
  'Button Group/Pink + Transparent': (
    <ButtonGroup>
      <Button color="pink">Button 1</Button>
      <Button color="transparent">Button 2</Button>
    </ButtonGroup>
  ),
  'Button Group/Pink + Transparent x2': (
    <ButtonGroup>
      <Button color="pink">Button 1</Button>
      <Button color="transparent">Button 2</Button>
      <Button color="transparent">Button 3</Button>
    </ButtonGroup>
  )
};
