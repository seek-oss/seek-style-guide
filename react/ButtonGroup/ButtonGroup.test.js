import React from 'react';
import { shallow } from 'enzyme';
import ButtonGroup from './ButtonGroup';
import Button from '../Button/Button';

describe('ButtonGroup:', () => {
  it('should render a button group with node', () => {
    const wrapper = shallow(
      <ButtonGroup>
        <Button color="blue">
          <h5>SEEK</h5>
        </Button>
      </ButtonGroup>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should combine className when passed in', () => {
    const wrapper = shallow(
      <ButtonGroup className="combineMe">
        <Button color="blue">
          <h5>SEEK</h5>
        </Button>
      </ButtonGroup>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
