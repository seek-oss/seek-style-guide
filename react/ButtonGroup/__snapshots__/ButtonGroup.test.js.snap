// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup: should combine className when passed in 1`] = `
<div
  className="combineMe Button__group"
>
  <Button
    className=""
    color="blue"
    component="button"
    fullWidth={false}
    ghost={false}
    loading={false}
  >
    <h5>
      SEEK
    </h5>
  </Button>
</div>
`;

exports[`ButtonGroup: should render a button group with node 1`] = `
<div
  className="Button__group"
>
  <Button
    className=""
    color="blue"
    component="button"
    fullWidth={false}
    ghost={false}
    loading={false}
  >
    <h5>
      SEEK
    </h5>
  </Button>
</div>
`;
